File: 1457. Pseudo-Palindromic Paths in a Binary Tree/pseudoPalindromicPaths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
    int ans = 0;
    void solve(TreeNode *node, vector<int> &count)
    {
        if (node == NULL)
            return;

        count[node->val]++;
        if (node->left == NULL && node->right == NULL)
        {
            int ones = 0;
            for (auto it : count)
            {
                if (it % 2 != 0)
                    ones++;
            }
            if (ones <= 1)
                ans++;
        }

        solve(node->left, count);

        solve(node->right, count);

        count[node->val]--;
    }
    int pseudoPalindromicPaths(TreeNode *root)
    {

        vector<int> count(10, 0);
        solve(root, count);
        return ans;
    }
};

File: 1457. Pseudo-Palindromic Paths in a Binary Tree/pseudoPalindromicPaths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    int ans = 0;

    public void solve(TreeNode node, int count) {
        if (node == null)
            return;

        count = count ^ (1 << node.val);
        if (node.left == null && node.right == null) {
            if ((count & (count - 1)) == 0)
                ans++;
        }

        solve(node.left, count);
        solve(node.right, count);
    }

    public int pseudoPalindromicPaths(TreeNode root) {
        solve(root, 0);
        return ans;

    }
}